src/test/java/de/unijena/cheminf/clustering/desccalc/DescriptorTest.java

```diff
@@ -108,83 +3413,452 @@ public void test_MOLECULAR_WEIGHT() throws Exception {
+
+        ElectronDonation[] models = {
+                Aromaticity.Model.Daylight,
+                Aromaticity.Model.CDK_2x,
+                Aromaticity.Model.CDK_1x,
+                Aromaticity.Model.CDK_AtomTypes,
+                Aromaticity.Model.Mdl,
+                Aromaticity.Model.OpenSmiles,
+                Aromaticity.Model.PiBonds
+        };
+
+        for (ElectronDonation model : models) {
+            // Parse fresh molecule for each model
+            IAtomContainer tmpMolecule = tmpSmilesParser.parseSmiles(tmpSmiles);
+
+            // Apply aromaticity with the current model
+            Descriptor.setAromaticity(tmpMolecule, model);
+
+            IAtomContainer[] tmpMoleculesArray = new IAtomContainer[]{tmpMolecule};
+            int tmpStartIndex = 0;
+            Descriptor[] tmpDescriptors = new Descriptor[]{Descriptor.LARGEST_PI_SYSTEM};
+            boolean tmpIsParallelCalculation = false;
+
+            try {
+                Assertions.assertEquals(1, Descriptor.getNumberOfComponents(tmpDescriptors));
+
+                float[][] tmpMatrix = new float[][]
+                        {
+                                {0f}
+                        };
+                Assertions.assertTrue(
+                        Descriptor.setDescriptorsForMoleculesByMoleculeParallelizationSynchronized(
+                                tmpDescriptors,
+                                tmpMoleculesArray,
+                                tmpMatrix,
+                                tmpStartIndex,
+                                tmpIsParallelCalculation
+                        )
+                );
+                Assertions.assertEquals(8, tmpMatrix[0][0]);
+
+                tmpMatrix = new float[][]
+                        {
+                                {0f}
+                        };
+                Assertions.assertTrue(
+                        Descriptor.setDescriptorsForMoleculesByMoleculeParallelizationNew(
+                                tmpDescriptors,
+                                tmpMoleculesArray,
+                                tmpMatrix,
+                                tmpStartIndex,
+                                tmpIsParallelCalculation
+                        )
+                );
+                Assertions.assertEquals(8, tmpMatrix[0][0]);
+
+                tmpMatrix = new float[][]
+                        {
+                                {0f}
+                        };
+                Assertions.assertTrue(
+                        Descriptor.setDescriptorsForMoleculesByDescriptorParallelization(
+                                tmpDescriptors,
+                                tmpMoleculesArray,
+                                tmpMatrix,
+                                tmpStartIndex,
+                                tmpIsParallelCalculation
+                        )
+                );
+                Assertions.assertEquals(8, tmpMatrix[0][0]);
```

Comment:
It is the same value for all models (8)? What would be the result if not aromaticity detection was performed beforehand? Also, pi system != aromatic system, so I would expect asromaticity to not play a role here.